client/src/components/RequestLunch.js

```diff
@@ -0,0 +1,276 @@
+		milk: false,
+		eggs: false,
+		peanuts: false,
+		fish: false,
+		soy: false,
+		wheat: false,
+	});
+
+	//states for checkboxes - dietary requirements
+	const [dietaryRequirements, setDietaryRequirements] = useState({
+		vegetarian: false,
+		vegan: false,
+		halal: false,
+		diabetic: false,
+	});
+
+	//states for text field - others restrictions
+	const [otherDietaryRestrictions, setOtherDietaryRestrictions] =
+		useState("");
+
+	//event for radio buttons - lunch request
+	const handleChangeInputRadioButtons = (event) => {
+		setLunchOption(event.target.value);
+	};
+
+	//event for dietary restrictions
+	const handleChangeInputRestrictions = (event) => {
+		const objectKey = event.target.name;
+		const objectValue = event.target.checked;
+		setDietaryRestrictions({
+			...dietaryRestrictions,
+			[objectKey]: objectValue,
+		});
+	};
+
+	//event for dietary requirements
+	const handleChangeInputRequirements = (event) => {
+		const objectKey = event.target.name;
+		const objectValue = event.target.checked;
+		setDietaryRequirements({
+			...dietaryRequirements,
+			[objectKey]: objectValue,
+		});
+	};
+
+	//event for text field
+	const handleChangeTextField = (event) => {
+		setOtherDietaryRestrictions(event.target.value);
+	};
+
+	//submit form event
+	const handleSubmit = () => {
+		fetch("http://localhost:3100/api/lunch/dietary", {
+			method: "POST",
+			headers: { "Content-Type": "application/json" },
+			body: JSON.stringify({
+				lunchOption: lunchOption,
+				dietaryRestrictions: dietaryRestrictions,
+				dietaryRequirements: dietaryRequirements,
+				otherDietaryRestrictions: otherDietaryRestrictions,
+			}),
+		})
+			.then((res) => res.json())
+			.then((data) => alert(data.msg))
+			.catch((err) => console.log(err));
+	};
+
+	return (
+		<Box sx={{ boxShadow: 3, mx: "auto", my: 6, p: 4, width: "80%" }}>
+			<Typography variant="h5" sx={{ mb: 2 }} align="center">
+				Lunch Requests
+			</Typography>
+			<FormControl>
+				{/* radio button for selecting if user wants/ doesn't want the meal to be provided*/}
+				<FormLabel id="lunch-radio-buttons-group-label">
+					Would you like to have the lunch provided by us?
+				</FormLabel>
+				<RadioGroup
+					aria-labelledby="lunch-radio-buttons-group-label"
+					name="lunch-radio-buttons-group"
+					value={lunchOption}
+					onChange={handleChangeInputRadioButtons}
+				>
+					<Box sx={{ display: "flex" }}>
+						<FormControlLabel value="yes" control={<Radio />} label="Yes" />
+						<FormControlLabel value="no" control={<Radio />} label="No" />
+					</Box>
+				</RadioGroup>
+
+				{lunchOption == "yes" ? (
+					<>
+						{/* checkboxes for selecting any dietary restrictions */}
+						<FormLabel
+							id="restrictions-radio-buttons-group-label"
+							component="legend"
+						>
+							Do you have any dietary restrictions?
+						</FormLabel>
+						<Box sx={{ display: "flex" }}>
+							<FormGroup>
+								<FormControlLabel
```

Comment:
**Stop Hard Coding**
from line 128 up-to 190 (almost 68 lines, which is to much ) Please,try to use some methods.